browser/src/slideshow/engine/SlideShowNavigator.ts

```diff
@@ -243,8 +261,62 @@ class SlideShowNavigator {
 	}
 
 	private clickHandler(aEvent: MouseEvent) {
-		if (aEvent.button === 0) this.dispatchEffect();
-		else if (aEvent.button === 2) this.switchSlide(-1, false);
+		if (aEvent.button === 0) {
+			const slideInfo = this.theMetaPres.getSlideInfoByIndex(this.currentSlide);
+			if (slideInfo.interactions.length == 0) {
+				this.dispatchEffect();
+				return;
+			}
+
+			// Get the coordinates of the click
+			const canvas = this.presenter.getCanvas();
+			const width = canvas.width;
+			const height = canvas.height;
+
+			const x = (aEvent.offsetX / width) * this.theMetaPres.slideWidth;
+			const y = (aEvent.offsetY / height) * this.theMetaPres.slideHeight;
+
+			const shape = slideInfo.interactions.find((shape) =>
+				hitTest(shape.bounds, x, y),
+			);
+			if (shape) {
+				this._onExecuteInteraction(shape.clickAction);
+			} else {
+				this.dispatchEffect();
+			}
+		} else if (aEvent.button === 2) {
+			this.switchSlide(-1, false);
+		}
+	}
+
+	_onExecuteInteraction(action: ClickAction) {
+		if (action) {
+			switch (action.action) {
+				case 'prevpage':
+					this.displaySlide(
```

Comment:
easier to call `this.switchSlide(-1, true)`;  `displaySlide` already takes care to check that the passed slide index is valid

Reply:
done